src/main.py

```diff
@@ -1,4 +1,44 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import sys
+
 from seedsigner.controller import Controller
 
-# Get the one and only Controller instance and start our main loop
-Controller.get_instance().start()
+logger = logging.getLogger(__name__)
+
+
+def main(sys_argv=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-l",
+        "--loglevel",
+        choices=list((logging._nameToLevel.keys())),
+        default="INFO",
+        type=str,
+        help=(
+            "Set the log level (default: %(default)s), WARNING: changing the log level "
+            "to something more verbose than %(default)s may result in unwanted data "
+            "being written to stderr"
+        ),
+    )
+
+    args = parser.parse_args(sys_argv)
+
+    root_logger = logging.getLogger()
+    root_logger.setLevel(logging.getLevelName(args.loglevel))
+    console_handler = logging.StreamHandler(sys.stderr)
+    console_handler.setFormatter(
+        logging.Formatter("%(asctime)s.%(msecs)03d %(levelname)s:\t%(message)s")
+    )
+    root_logger.addHandler(console_handler)
+
```

Comment:
PIL gets noisy when the logging level is DEBUG across the board.

There are tons of ways to configure more specific per-module log levels. I wasn't happy with my `logging.ini` experiments, so without re-inventing the wheel too much (or fussing too much going further down the logging rabbit hole), a simple dict at the start of `main.py` could suffice.

```python
DEFAULT_MODULE_LOG_LEVELS = {
    "PIL": logging.WARNING,
    # "seedsigner.gui.toast": logging.DEBUG,  # example of more specific submodule logging config
}
```

And then after the `root_logger` is configured:
```python
    # Set log levels for specific modules
    for module, level in DEFAULT_MODULE_LOG_LEVELS.items():
        logging.getLogger(module).setLevel(level)
```

Reply:
added via https://github.com/SeedSigner/seedsigner/pull/558/commits/af2cdf80369d67d3053046ef649fb74c28ff4f67 :)